wd_helper.au3

```diff
@@ -1037,7 +1080,12 @@ Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand)
 		__WD_ConsoleWrite($sFuncName & ': ' & ((IsArray($vResult)) ? "(array)" : $vResult) & @CRLF)
 	EndIf
 
-	Return SetError(__WD_Error($sFuncName, $iErr), $_WD_HTTPRESULT, $vResult)
+	Local $iExt = $_WD_HTTPRESULT
+
+	; if $iSelectedOption was found then pass the ID ($iSelectedOption) to @extended
+	If $_WD_HTTPRESULT = $HTTP_STATUS_OK And $iSelectedOption <> -1 Then $iExt = $iSelectedOption
```

Comment:
Not sure why you are checking $_WD_HTTPRESULT here. Wouldn't it be more appropriate to check $iErr?

Reply:
Done in commit `$iExt = $iSelectedOption`

Reply:
The meaning of @extended shouldn't vary within the function. If you want to return the "selected" status, then this should be done for both "options" routines. Have you considered adding another column to the array?

Reply:
For example what type of column ?

Reply:
Boolean is my initial thought

Reply:
For selected ? 
If so I prefer to return them in EXT as you have it direclty an you do not must process them outside the function to check which one is selected.

Reply:
Please keep the following in mind --

- This function wasn't designed to return the element's selected status
- The current "options" implementation could be modified so that it is both reliable and fast

One option, which I actually like, is to modify _WD_ElementSelectAction to support a "selected" command

Reply:
> One option, which I actually like, is to modify _WD_ElementSelectAction to support a "selected" command

Ok I will provide such option soon.

Reply:
You may want to submit it as a separate PR so it doesn't get held back by this one

Reply:
Ok

Reply:
I'm no longer convinced that adding a "selected" command is a good thing. I believe the current "value" command already gives us the selected item's value. So what purpose would "selected" command serve?

Reply:
So the descriptions 
```AutoIt
; |VALUE - Retrieve current value
```
could be more adequate, something like
```AutoIt
; |VALUE - Retrieve current selected value
```
or
```AutoIt
; |VALUE - Retrieve current value - selected item
```

Reply:
> So what purpose would "selected" command serve?

None.

Reply:
> Please keep the following in mind --
> 
> * This function wasn't designed to return the element's selected status
> * The current "options" implementation could be modified so that it is both reliable and fast
> 
> One option, which I actually like, is to modify _WD_ElementSelectAction to support a "selected" command

So what we do now with this ?
Do you mean that `OPTIONS2` should not return the selected value in EXTENDED ?

Reply:
We may not need to include an `Options2` if a suitable replacement for the existing `Options` is created. Here's my initial attempt --

```autoit
 Func _WD_ElementSelectAction2($sSession, $sSelectElement, $sCommand)
	Local Const $sFuncName = "_WD_ElementSelectAction2"
	Local $sNodeName, $sJsonElement, $vResult
	Local $sText, $aOptions

	$sNodeName = _WD_ElementAction($sSession, $sSelectElement, 'property', 'nodeName')
	Local $iErr = @error

	If $iErr = $_WD_ERROR_Success And $sNodeName = 'select' Then

		Switch $sCommand
			Case 'value'
				; Retrieve current value of designated Select element
				$sJsonElement = __WD_JsonElement($sSelectElement)
				$vResult = _WD_ExecuteScript($sSession, "return arguments[0].value", $sJsonElement, Default, $_WD_JSON_Value)
				$iErr = @error

			Case 'options'
				; Retrieve array containing value / label attributes from the Select element's options
				Local $sScript = "var result =''; var options = arguments[0].options; for (let i = 0; i < options.length; i++) {result += options[i].value + '|' + options[i].label + '\n'} return result;"
				Local $sText = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelect), Default, $_WD_JSON_Value)
				$iErr = @error

				$sText = StringStripWS($sText, $STR_STRIPTRAILING)
				Local $aOut[0][2]
				_ArrayAdd($aOut, $sText, 0, Default, @LF, 1)
				$vResult = $aOut
			Case Else
				Return SetError(__WD_Error($sFuncName, $_WD_ERROR_InvalidDataType, "(Value|Options) $sCommand=>" & $sCommand), 0, "")

		EndSwitch
	Else
		$iErr = $_WD_ERROR_InvalidArgue
	EndIf

	If $_WD_DEBUG = $_WD_DEBUG_Info Then
		__WD_ConsoleWrite($sFuncName & ': ' & ((IsArray($vResult)) ? "(array)" : $vResult) & @CRLF)
	EndIf

	Return SetError(__WD_Error($sFuncName, $iErr), $_WD_HTTPRESULT, $vResult)
EndFunc   ;==>_WD_ElementSelectAction2
```

Would be interesting to see how the speed compares to your version.

Reply:
> We may not need to include an `Options2` if a suitable replacement for the existing `Options` is created. Here's my initial attempt --

yes

> Would be interesting to see how the speed compares to your version.

Durations = 8.855816 sec : OLD
Durations = 0.041332 sec : HTML PROCESSING
Durations = 0.0332274 sec : JAVA SCRIPT

Reply:
Look at my recent commits.
btw. this conversation could be resolved.